src/components/Activities/Activity3/Game/TimeAllocation.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useContext, useEffect} from 'react';
+import { GameContext } from '.';
+
+import Clock from '../../../../assets/activity3/game/Clock.svg';
+import Debug from '../../../../assets/activity3/game/Debug.svg';
+import Graph from '../../../../assets/activity3/game/Graph.svg';
+import Hammer from '../../../../assets/activity3/game/Hammer.svg';
+
+import NumberSelection from './NumberSelection';
+import { TASKS, LOW_DAY_THRESHOLD, HIGH_DAY_THRESHOLD } from './GameConstants';
+
+interface TimeAllocationProps {
+  setTaskSelection: (variables: number[]) => void;
+  initialTasks: number[];
+}
+
+function TimeAllocation(props: TimeAllocationProps): JSX.Element {
+  const {daysLeft, setDaysLeft, goNextState} = useContext(GameContext);
+  const { setTaskSelection, initialTasks } = props;
+  const { BUILD, DEBUG, ABTEST } = TASKS;
+
+  // initialize allocations based on props
+  const initAllocation = (task : number) => {
+    return initialTasks[task] ? initialTasks[task] : 0;
+  };
+  const [buildDays, setBuildDays] = useState(() => initAllocation(BUILD));
+  const [debugDays, setDebugDays] = useState(() => initAllocation(DEBUG));
+  const [testDays, setTestDays] = useState(() => initAllocation(ABTEST));
+
+  useEffect(() => {
+    setDaysLeft && setDaysLeft(daysLeft ? (daysLeft + buildDays + debugDays + testDays) : 0);
+  }, []);
```

Comment:
figure out the weird state onmount stuff for you! 
```suggestion
  const [buildDays, setBuildDays] = useState(0);
  const [debugDays, setDebugDays] = useState(0);
  const [testDays, setTestDays] = useState(0);

  useEffect(() => {
    const initBuildDays = initialTasks[BUILD] ?? 0;
    const initDebugDays = initialTasks[DEBUG] ?? 0;
    const initABTestDays = initialTasks[ABTEST] ?? 0;
    setBuildDays(initBuildDays);
    setDebugDays(initDebugDays);
    setTestDays(initABTestDays);
  }, []);
```

Reply:
see below about `??`